src/vmm/src/persist.rs

```diff
@@ -280,6 +275,29 @@ fn snapshot_memory_to_file(
     }
 }
 
+/// Validate the microVM version and translate it to its corresponding snapshot data format.
+pub fn get_snapshot_data_version(
+    version: &Option<String>,
+    version_map: &VersionMap,
+    #[cfg(target_arch = "x86_64")] device_manager: &MMIODeviceManager,
```

Comment:
Since in the future we might want to look at other vmm components for validating the microVM version, I would suggest to use as argument to this function Vmm instead of MMIODeviceManager. Also, you could use _vmm to avoid as much as possible the use of cfg's:
```
pub fn get_snapshot_data_version(
    version: &Option<String>,
    version_map: &VersionMap,
    _vmm: &Vmm,
) -> std::result::Result<u16, CreateSnapshotError> {
```

Reply:
Ack.